### PR TITLE
[Consensus] Add Quorum Certificate to the block

### PR DESCRIPTION
**Pull request by gokutheengineer**
## Description

Persisted blocks doesn't contain QC in block header field. When doing state sync, peers need the QC to validate the block. 

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [X] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Add quorum certificate to the block before committing in leader and replica
- Add new error `ErrNoQcInReceivedBlock` 

## Testing

- [X] `make develop_test`
- [X] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [X] I have tested my changes using the available tooling
- [X] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: consensus/doc/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.0.0.37] - 2023-03-17
+
+- Add quorum certificate to the block before commiting to the persistence
```
Comment (Olshansk) [inline consensus/doc/CHANGELOG.md:12]:
```suggestion
- Add quorum certificate to the block before committing to persistence
```

Consider downloading grammarly.com

File: consensus/hotstuff_leader.go [hunk 2]
```diff
@@ -243,6 +243,13 @@ func (handler *HotstuffLeaderMessageHandler) HandleCommitMessage(m *consensusMod
 	}
 	m.broadcastToValidators(decideProposeMessage)
 
+	commitQcBytes, err := codec.GetCodec().Marshal(commitQC)
```
Comment (Olshansk) [inline consensus/hotstuff_leader.go:246]: Nice

**Review by Olshansk — CHANGES_REQUESTED** [review]
This is a great change. My only request: can you add:

1. An empty test with a TODO to add more quorum cert validation
2. Inside of one of the existing tests where we do `require.Equal(t, uint64(1), getBlockRes.Block.GetBlockHeader().Height)`, why not validate the quorum cert dat as well?

**Comment by gokutheengineer** [comment]
> 1. An empty test with a TODO to add more quorum cert validation
Done. 

> 2. Inside of one of the existing tests where we do `require.Equal(t, uint64(1), getBlockRes.Block.GetBlockHeader().Height)`, why not validate the quorum cert dat as well?
Currently persistence mock returns an almost empty whit only block height is set in blockheads. Therefore, in this PR we can't do QC validation. However, with PR https://github.com/pokt-network/pocket/pull/594, we will be able to do QC verification, as I updated test suite to generate mock blocks with valid QCs.  Wdyt?

File: consensus/state_sync/module.go [hunk 3]
```diff
@@ -131,6 +131,11 @@ func (m *stateSync) HandleGetBlockResponse(blockRes *typesCons.GetBlockResponse)
 
 	m.logger.Info().Fields(fields).Msgf("Received GetBlockResponse: %s", blockRes)
 
+	if blockRes.Block.BlockHeader.GetQuorumCertificate() == nil {
+		m.logger.Error().Err(typesCons.ErrNoQcInReceivedBlock)
```
Comment (Olshansk) [inline consensus/state_sync/module.go:135]: Add a message too

**Review by Olshansk — CHANGES_REQUESTED** [review]

**Comment by Olshansk** [comment]
@gokutheengineer It seems that you merged in these changes in #575, even though they're being reviewed here. This is not a big problem because these are small changes, but please be more careful about this in the future.

File: consensus/e2e_tests/hotstuff_test.go [hunk 4]
```diff
@@ -236,6 +238,10 @@ func TestHotstuff4Nodes1BlockHappyPath(t *testing.T) {
 
 // TODO: Implement these tests and use them as a starting point for new ones. Consider using ChatGPT to help you out :)
 
+func TestValidateQuorumCertificate(t *testing.T) {
+	t.Skip()
+}
```
[the remark this answers is not in the document]
Comment (gokutheengineer) [reply consensus/e2e_tests/hotstuff_test.go]: I added few more. I will be thinking about other potential test cases and will be updating this.

**Review by Olshansk — APPROVED** [review]

**Pull request merged**